src/server/storage_engine/block_device/allocator.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 labring. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use libc::ioctl;
+use nix::fcntl::{open, OFlag};
+use parking_lot::Mutex;
+
+use crate::server::EngineError;
+
+//#define BLKGETSIZE _IO(0x12,96)	/* return device size /512 (long *arg) */
+const BLOCKGETSIZE: u64 = 0x1260;
+
+pub const CHUNK: u64 = 512 * 8;
+const SECTOR: u64 = 512;
+
+pub(crate) trait Allocator {
+    fn new(path: &str) -> Self;
+    fn allocator_space(&self, lenth: u64) -> u64;
+}
+
+/*
+ * This Allocator use for memory.
+ */
+#[allow(unused)]
+pub(crate) struct BitmapAllocator {
+    block_space: Arc<Mutex<u64>>,
```

Comment:
AtomicU64

Reply:
After testing, the performance of atomic variables is not better, and the competition is very serious. This part will be optimized later, using mutex temporarily

Reply:
How do you test?

Reply:
rust Criterion， watch time